BEP5.md

```diff
@@ -0,0 +1,275 @@
+
+## 2.  Abstract
+
+The BEP-5 Proposal describes a common set of rules for adding token resources to BEP-2 tokens on the Binance Chain ecosystem. It adds the following detail of a token on Binance Chain:
+
+- What information can be added to a BEP-2 Token (BEP-5 Token Resource)
+- What actions can be performed to manage BEP-5 Token Resources on Binance Chain
+
+## 3.  Status
+
+This BEP is a work in progress (WIP).
+
+## 4.  Motivation
+
+BEP-2 tokens currently specify their `Name`, `Address`, `Symbol`, `Supply` and `Mintable` status. However, no other information can be added, which results in either Binance or wallet developers hosting additional information that is not distributed widely across the ecosystem and can become siloed.
+
+> As an example, icons for tokens allow easier identification of tokens in wallets and dapps, and should be sourced from somewhere. Currently developers host their own copies of token logos, or retrieve them from CoinMarketCap paid API.
+
+In order to facilitate a rich ecosystem of tokens and the standardising of information around them, tokens should be allowed to add a resource pointer that allows retrieving of basic (and extensible) token information.
+
+The resource pointer will be a Uniform Resource Identifier (URI) that queries for resources.
+
+## 5.  Specification
+
+### 5.1 Token Resource
+
+BEP-2 Tokens can be deployed to Binance Chain via the BEP-2 token issuing process.
+
+A token resource can be added as part of the listing command:
+
+```
+- Source Address: Source Address is the owner of the issued token.
+
+- Token Name: Token Name represents the long name of the token - e.g. "MyToken".
+
+- Symbol: Symbol is the identifier of the newly issued token.
+
+- Total Supply: Total supply will be the total number of issued tokens.
+
+- Mintable: Mintable means whether this token can be minted in the future, which would increase the total supply of the token
+
+- URI: Token URI returns a URL that specifies Token Resources
+
+```
+
+An example of a URI:
+
+```
+URI: "https://token.io/token_data.json"
+```
+
+
+### 5.2 URI Specification
+
+The Token URI points to a URL that is hosted by project teams that returns a JSON file:
+
+```json
+{
+  "description": "Token is a decentralised project that is building the future of payments",
+  "website": "https://token.io",
+  "icon": "https://token.io/icon.png",
+  "contact": {
+    "website": "https://token.io",
+    "twitter": "https://twitter.com/token",
+    "telegram": "https://t.me/token",
+    "support": "https://support.token.io/",
+    "blog": "https://medium.com/token"
+  },
+  "media": {
+    "icon": "https://token.io/icon.png",
+    "icon_large": "https://token.io/icon_large.png",
+    "logo": "https://token.io/logo.png",
+    "logo_large": "https://token.io/logo_large.png",
+  },
+  "other": {
+    "****": "****",
+  }
+}
+```
+
+### 5.3 JSON SCHEMA
```

Comment:
This data structure is quite complicated and not easy for adopting future changes.

Reply:
Hi @huangsuyu 

To clarify, the only addition to the protocol is the following field which needs to be added to the token and is less than 120 bytes:

```
	URI: "https://token.io/token_data.json"   <--- 120byte string
```

The URI can be changed/updated at any time. This makes it very easy for adopting future changes. 

The client (DEX/wallet/DAPP) then retrieves the json packet and displays the data. This data is off-chain and is *very* flexible. 

As a minimum standard, all that needs to be added are the following fields:

- description
- icon
- website

Everything else is optional but enhances the DEX/Wallet/DAPP, and can be updated at any time. 

To sum up:

On-chain: 120byte string
Off-chain queried by clients: json packet

Please let me know if you have more questions.